### PR TITLE
Add UKPA membership to user profile

### LINKED ISSUE #71: Add UKPA Membership
When a user has purchased a ticket, their profile should include whether they wish to take up their entitlement to membership of the UKPA.

If so, they need to confirm that they agree that they are aware of the Duty of Members described in Section 9.3 of the UKPA constitution, which states:

> It is the duty of each member of the UKPA to exercise his or her powers as a member of the UKPA in the way that he or she decides in good faith would be most likely to further the purposes of the UKPA.

### PR DESCRIPTION
**Pull request by meatballs**
Fixes #71

### CODE REVIEW

File: accounts/forms.py [hunk 1]
```diff
@@ -65,6 +65,7 @@ class Meta:
             'childcare_reqs',
             'dietary_reqs_yn',
             'dietary_reqs',
+            'ukpa_membership',
```
Comment (inglesp) [inline accounts/forms.py]: Could you rename this to `is_ukpa_member`?

File: accounts/templates/accounts/edit_profile.html [hunk 2]
```diff
@@ -72,6 +72,26 @@
...
+        <p>You can read more details about the UKPA and becoming a member on the <a href="http://2017.pyconuk.org/ukpa/">conference website</a></p>
+        <table class="table">
+          <tr>
+            <th class="col-md-4">Do you wish to become a member of the UKPA?</th>
+            <td>
+              <div class="btn-group" data-toggle="buttons">
+                <label class="btn btn-default{% if form.ukpa_membership.value == True %} active{% endif %}">
+                  <input type="radio" name="ukpa_membership" value="2" {% if form.ukpa_membership.value == True %} checked{% endif %}>Yes
+                </label>
+                <label class="btn btn-default{% if form.ukpa_membership.value == True %} active{% endif %}">
```
Comment (inglesp) [inline accounts/templates/accounts/edit_profile.html]: This should test whether `form.ukpa_membership.value == False`.

File: accounts/templates/accounts/profile.html [hunk 3]
```diff
@@ -60,6 +60,24 @@
...
     <div>
+      <h3>UK Python Association</h3>
+        <table class='table'>
+          <tr>
+            <th class="col-md-4">Membership</th>
+            <td>
+              {% if user.ukpa_membership == None %}
+              unknown
+              {% elif user.ukpa_membership %}
+              {{ user.ukpa_membership }}
```
Comment (inglesp) [inline accounts/templates/accounts/profile.html]: Currently this shows "True" or "none".  Could you change it to "Yes" or "No"?

File: reports/views.py [hunk 4]
```diff
@@ -67,6 +68,19 @@ def get_context_data(self):
...
+    def get_context_data(self):
+        members = User.objects.filter(ukpa_membership=True)
+        rows = [[member.name, member.email_addr] for member in members]
+
+        return {
+            'title': self.title,
+            'headings': ['Name', 'email'],
+            'rows': rows,
+        }
+
```
Comment (inglesp) [inline reports/views.py]: I'm glad this was (apparently!) self-explanatory.

**Review by inglesp — CHANGES_REQUESTED** [review]
Hi @meatballs -- looks pretty good.

There's a single failing test, which is because the user returned by `accounts.tests.factories.create_user_with_full_profile` doesn't have the `ukpa_membership` attribute set.  This should be an easy fix.  (Running the tests locally should be a matter of `./manage.py test`, and if it's not, please let me know.)

**Comment by meatballs** [comment]
I think the test failures are now just the ones already fixed by #77

**Review by inglesp — CHANGES_REQUESTED** [review]
Sorry, something else that's occurred to me: we only want to offer membership to people who have a ticket for the conference.

We can use `User.get_ticket()` to test whether a user has a ticket (it'll return `None` if they don't), and use this to decide whether display the `is_ukpa_member` field on the `edit_profile.html` form.

We should also use whether a user has a ticket or not to determine what `User.profile_complete()` returns.

File: accounts/forms.py [hunk 5]
```diff
@@ -88,6 +88,9 @@ def __init__(self, *args, **kwargs):
             if value is not None and value not in choices:
                 widget.choices.insert(1, [value, value])
 
+        if self.instance.get_ticket() is None:
+            self.fields['is_ukpa_member'].disabled = True
```
Comment (inglesp) [inline accounts/forms.py]: Could you add a comment here?  I don't understand what setting `disabled` is for.

**Review by inglesp — CHANGES_REQUESTED** [review]

File: accounts/forms.py (hunk 5 above)
Comment (meatballs) [inline accounts/forms.py]: I read somewhere that it prevents the field being used even if somebody attempts to do so by sending a post request directly to the server.

**Review by inglesp — APPROVED** [review]

**Pull request merged**